WordPress/src/test/java/org/wordpress/android/ui/accounts/login/LoginNoSitesViewModelTest.kt

```diff
@@ -0,0 +1,192 @@
+package org.wordpress.android.ui.accounts.login
+
+import android.os.Bundle
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.argThat
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.InternalCoroutinesApi
+import org.assertj.core.api.Assertions
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mock
+import org.wordpress.android.BaseUnitTest
+import org.wordpress.android.TEST_DISPATCHER
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.AccountModel
+import org.wordpress.android.fluxc.store.AccountStore
+import org.wordpress.android.ui.accounts.LoginNavigationEvents
+import org.wordpress.android.ui.accounts.LoginNavigationEvents.ShowInstructions
+import org.wordpress.android.ui.accounts.LoginNavigationEvents.ShowSignInForResultJetpackOnly
+import org.wordpress.android.ui.accounts.UnifiedLoginTracker
+import org.wordpress.android.ui.accounts.login.LoginNoSitesViewModel.State.NoUser
+import org.wordpress.android.ui.accounts.login.LoginNoSitesViewModel.State.ShowUser
+import org.wordpress.android.ui.accounts.login.LoginNoSitesViewModel.UiModel
+
+@InternalCoroutinesApi
+class LoginNoSitesViewModelTest : BaseUnitTest() {
+    @Mock lateinit var unifiedLoginTracker: UnifiedLoginTracker
+    @Mock lateinit var wordPress: WordPress
+    @Mock lateinit var accountStore: AccountStore
+    @Mock lateinit var savedInstanceState: Bundle
+    private lateinit var viewModel: LoginNoSitesViewModel
+
+    @Before
+    fun setUp() {
+        viewModel = LoginNoSitesViewModel(
+                unifiedLoginTracker,
+                accountStore,
+                TEST_DISPATCHER,
+                TEST_DISPATCHER
+        )
+    }
+
+    @Test
+    fun `given started, then application sign out is invoked`() {
+        setupAccountItem()
+
+        startViewModel()
+
+        verify(wordPress).wordPressComSignOut()
+    }
+
+    @Test
+    fun `given savedInstanceState is NoUser, when started, then state is recreated as NoUser`() {
```

Comment:
1) Suggestion (💡): I would avoid using code specific language on test sentences, like classes or camel-case function references.
2) Question (❓): Since the `setupInstanceStateForNoUser()` and `setupInstanceStateForShowUser()` are only used once and are actually one-liners (ish), maybe extracting those to a method might convey that they are not. Did you do that just so that the setup becomes more obvious to the reader?
3) Minor (🔍): Consider moving the `Assertions.` to imports.

Reply:
> Did you do that just so that the setup becomes more obvious to the reader?

TBH, I wrote those methods first and then they only got used once, but I left it because it's clearer to read the test with them then without.